src/Context/context/Authcontext.jsx

```diff
@@ -0,0 +1,108 @@
+import axios from "axios";
+import { createContext, useContext, useState, useReducer } from "react";
+import { useLocation, useNavigate } from "react-router-dom";
+import { Toast } from "../../components";
+import { authReducer } from "./authReducer";
+
+const authcontext = createContext();
+
+const AuthProvider = ({ children }) => {
+  const [userDetailes, setUserDetailes] = useState({
+    token: localStorage.getItem("token") || "",
+    user: JSON.parse(localStorage.getItem("users")) || {},
+  });
+  const navigate = useNavigate();
+  const location = useLocation();
+  const [authState, Authdispatch] = useReducer(authReducer, {
+    email: "",
+    firstName: "",
+    lastName: "",
+    password: "",
+    confirPassword: "",
+    isSubmit: false,
+  });
+
```

Comment:
I don't think these details are needed to be stores in context as they wont be needed everytime and you are already storing the user details in local storage. You can define a local state variable for this in SignUp component itself.

Reply:
yeah, i will change it.thank you for pointing out